packages/dropdown/src/react/index.js

```diff
@@ -157,9 +158,33 @@ const Dropdown = React.forwardRef((props, forwardedRef) => {
       setKeyboarding(false)
     }
   }
+
   useEffect(() => {
-    isOpen && document.addEventListener('click', handleClickOutsideMenu)
-    return () => document.removeEventListener('click', handleClickOutsideMenu)
+    let currentAnimationFrame = null
+    const requestAnimationFrame = () => {
+      window.cancelAnimationFrame(currentAnimationFrame)
+      currentAnimationFrame = window.requestAnimationFrame(() => setOpen(false))
+      return currentAnimationFrame
+    }
+    let timeout = null
+    const clear = () => {
+      clearTimeout(timeout)
+      timeout = setTimeout(() => setOpen(false), 50)
+    }
+    if (isOpen) {
+      document.addEventListener('click', handleClickOutsideMenu)
+      document.addEventListener('contextmenu', handleClickOutsideMenu)
```

Comment:
i don't think we've closed on right click before and i'm not sure if that's correct. my rule of thumb has been to close on events that change focus(and the focus should be returned to the prev active element). a right click doesn't change focus so i'm unsure that this change is correct

Reply:
I just cloned the interaction behavior of a native select element here. It's probably not necessary.